probable discovery client bug fix + beginning mock integration tests to verify these scenarios.

govin opened the pull request.

cgc commented on discovery.js:
```diff
@@ -254,6 +256,7 @@ DiscoveryClient.prototype._singleAnnounce = function (announcement, cb) {
     body: announcement
   }, function (error, response, body) {
     if (error) {
+      disco.dropServer(server);
```
can we introspect on the type of the error to understand if the server should be dropped? so maybe a 502 or ECONNRESET should result in a drop, but something like a timeout shouldn't? but I guess if it's a good server, it'll be added back by the discovery response?

govin replied:
yeah figured if its a good server, it will be added back again. so didn't bother to inspect the type of error. would you like have that check?

cgc replied:
As an example, here's an HTTP request to an unbound port:
```javascript
> require('request')({url: 'http://localhost/'}, function(error) { console.log(error) });null
null
> { [Error: connect ECONNREFUSED]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect' }
```

cgc replied:
given that the server will be added again, I'd say we're fine.

stevenschlansker replied:
Maybe we could just log the error?

cgc commented on mock-integration-tests/updatesTest.js:
```diff
@@ -0,0 +1,164 @@
+										"staticAnnouncement":false,
+										"announceTime":"2015-03-30T18:26:52.178Z",
+										"serviceType":"discovery",
+										"serviceUri":DISCOVERY_SERVER_URLS[0]
```
prefer 2-space indentation.

cgc commented:
consider adding tests of failure scenarios?

govin commented:
@cgc yeah, definitely, just starting to build out the integration tests. will add failure scenarios.

cgc commented on tests/announceTests.js:
```diff
@@ -0,0 +1,120 @@
+			});
+		});
+
+		setTimeout(function() { 
```
consider using `disco.onUpdate`?

cgc commented on tests/announceTests.js:
```diff
@@ -0,0 +1,120 @@
+			});
+		});
+
+		setTimeout(function() { 
```
(duplicate comment) consider using `disco.onUpdate`?

stevenschlansker commented on tests/announceTests.js:
```diff
@@ -0,0 +1,120 @@
+	beforeEach(function(done){
+		nock.cleanAll();
+		nock.disableNetConnect();
+
```
Let's keep tabs consistent - the rest of the project uses 2-space tabs, and it looks like this file uses hard tabs?

stevenschlansker commented:
Looks pretty good.  Things outstanding to fix:
* Consistent tabbing style
* Add newlines to end of file (notice the little red error marker in the diff)
* Replace setTimeout usage with appropriate callbacks from discovery client (Time-based tests are inherently unreliable)

govin commented:
1) Converted tabs to spaces, removed newline at end of file.
2) On setTimeout(), I'm already using callbacks (onUpdate() callback) from the discovery client. The setTimeout() only exists as a double-check.
